src/smolvm/vm.py

```
@@ -1468,6 +1471,63 @@ def _maybe_enable_vsock(self, config: VMConfig, backend: str, vm_info: VMInfo) -
+            except NetworkError:
+                continue
+        raise NetworkError("No vsock CIDs available in pool")
+
+    @staticmethod
+    def _live_qemu_vsock_cids() -> set[int]:
+        """Return QEMU vhost-vsock CIDs visible in local process arguments."""
+        if platform.system() != "Linux":
+            return set()
+        try:
+            result = subprocess.run(
+                ["ps", "-eo", "args="],
```

Ruff (0.15.15)

[error] Starting a process with a partial executable path

(S607)